### PR TITLE
Added IEnumerable support for serialization and deserialization + unit tests.

### PR DESCRIPTION
**Pull request by YPermitin**
- Added support for IEnumerable types.
- Added unit test for new features

I need fresh look to my solution.
I am open for any suggestion and kicks to right direction :)

### CODE REVIEW

File: src/BinaryFormatter/BinaryConverter.cs [hunk 1]
```diff
@@ -27,16 +28,24 @@ public class BinaryConverter
...
 
         public byte[] Serialize(object obj)
         {
             Type t = obj.GetType();
+
             BaseTypeConverter converter;
             if (_converters.TryGetValue(t, out converter))
             {
                 return converter.Serialize(obj);
+            } else if (obj as IEnumerable != null)
```
Comment (lukasz-pyrzyk) [inline src/BinaryFormatter/BinaryConverter.cs]: maybe
```
obj is IEnumerable 
```
?

File: src/Tests/BinaryFormatterTests/TypeConverter/IEnumerableConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            simpleCollection.Add("Кто не ходит, тот и не падает.");
+            simpleCollection.Add(uint.MaxValue);
+            simpleCollection.Add(ulong.MaxValue);
+            simpleCollection.Add(ushort.MaxValue);
+
+            byte[] bytesSimpleCollection = converter.Serialize(simpleCollection);
+            var valueFromBytesSimpleCollection = converter.Deserialize(bytesSimpleCollection);
+            Assert.Equal(valueFromBytesSimpleCollection, simpleCollection);
+
+            List<object> complexCollection = new List<object>();
```
Comment (lukasz-pyrzyk) [inline src/Tests/BinaryFormatterTests/TypeConverter/IEnumerableConverterTests.cs]: I would prefer creating two different tests or test cases

**Comment by lukasz-pyrzyk** [comment]
and could you update readme? :))

File: src/BinaryFormatter/BinaryConverter.cs (hunk 1 above)
Comment (YPermitin) [inline src/BinaryFormatter/BinaryConverter.cs]: Fixed it.

File: src/Tests/BinaryFormatterTests/TypeConverter/IEnumerableConverterTests.cs (hunk 2 above)
Comment (YPermitin) [inline src/Tests/BinaryFormatterTests/TypeConverter/IEnumerableConverterTests.cs]: Ok, I splited tests.

**Comment by YPermitin** [comment]
@lukasz-pyrzyk Updating Readme was most difficult part =)

**Comment by lukasz-pyrzyk** [comment]
Thanks for great contribution. I will release new version today

**Pull request merged**